stablehlo/tests/infer_stablehlo.mlir

```diff
@@ -1037,6 +1036,14 @@ func.func @pad(%arg0: tensor<?x48x48x32xf32>) -> tensor<4xindex> {
 
 // -----
 
+// CHECK-LABEL: func @cholesky_bounds
+func.func @cholesky_bounds(%input: tensor<2x?x?xf32, #stablehlo.type_extensions<bounds = [?, 5, ?]>>) -> tensor<2x?x?xindex> {
+  %0 = "stablehlo.cholesky"(%input) { lower = true } : (tensor<2x?x?xf32, #stablehlo.type_extensions<bounds = [?, 5, ?]>>) -> tensor<2x?x?xf32>
```

Comment:
Both are fine. I learned this (tensor<*xf32>) in from https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/infer_stablehlo.mlir#L664
Personally I prefer this way because 1) it's shorter, 2) we return tensor<*xf32> from the op we are testing, and then in `hlo_test_infer.return_types` it's more clear that the test expect a very concrete type in "type0 = tensor<some_concrete_shape_with_bound>.

Reply:
I also noticed that some tests omit `%1 = "hlo_test_infer.return_types"(%0) {...} : (tensor<*xf32>) -> tensor<*xindex>` boilerplate in the CHECK directive, which is also a good idea.